### PR TITLE
Dev/logic

### PR DESCRIPTION
**Pull request by fvalela1**
Working app. Waiting on square root to be turned into 'undo' button to continue

### CODE REVIEW

File: app/src/main/java/com/example/francescovalela/androidtutorialwinter2018/Calculator.kt [hunk 1]
```diff
@@ -1,30 +1,94 @@
...
     val formula: Stack<FormulaItem> = Stack()
 
     fun pushOperand(operand: Double): Unit {
-        // TODO
+        formula.push(FormulaItem(operand = operand))
     }
 
     fun pushOperator(operator: Char): Unit {
-        // TODO
+        if (formula.isEmpty())
```
Comment (jacobrs) [inline app/src/main/java/com/example/francescovalela/androidtutorialwinter2018/Calculator.kt]: do you really need this?
Wouldn't
```kotlin
if (!formula.peek().isOperand) {
  formula.pop()
  formula.push(FormulaItem(operator = operator))
```
Be enough?

File: app/src/main/java/com/example/francescovalela/androidtutorialwinter2018/Calculator.kt [hunk 2]
```diff
@@ -1,30 +1,94 @@
...
+        var total = 0.0
+        var mostRecentOperator: Char? = null
+
+        for (item in formula) {
+            if (!item.isOperand)
+                mostRecentOperator = item.operator
+            else if (mostRecentOperator == null)
+                total += item.operand!!
+            else {
+                when (mostRecentOperator) {
```
Comment (jacobrs) [inline app/src/main/java/com/example/francescovalela/androidtutorialwinter2018/Calculator.kt:34]: Nice!

**Review by jacobrs — APPROVED** [review]
Looks good 👍

File: app/src/main/java/com/example/francescovalela/androidtutorialwinter2018/Calculator.kt (hunk 1 above)
Comment (fvalela1) [reply app/src/main/java/com/example/francescovalela/androidtutorialwinter2018/Calculator.kt]: Absolutely. New PR incoming.

**Pull request merged**